src/actions/lendingActions.js

```diff
@@ -0,0 +1,196 @@
+      supportedAssets,
+    );
+    dispatch(setDepositedAssetsAction(depositedAssets));
+  };
+};
+
+export const fetchDepositedAssetAction = (symbol: string) => {
+  return async (dispatch: Dispatch, getState: GetState) => {
+    const {
+      assets: { supportedAssets },
+      accounts: { data: accounts },
+      lending: { depositedAssets, isFetchingDepositedAssets },
+    } = getState();
+    const currentAccountAssets = accountAssetsSelector(getState());
+    const smartWalletAccount = findFirstSmartAccount(accounts);
+    if (!smartWalletAccount) return;
+
+    if (isFetchingDepositedAssets) return;
+    dispatch({ type: SET_FETCHING_LENDING_DEPOSITED_ASSETS });
+
+    const asset = getAssetData(getAssetsAsList(currentAccountAssets), supportedAssets, symbol);
+    const accountAddress = getAccountAddress(smartWalletAccount);
+
+    const updatedDepositedAsset = await aaveService.fetchAccountDepositedAsset(accountAddress, asset);
+    if (!updatedDepositedAsset) return;
+
+    const updatedDepositedAssets = depositedAssets.reduce((
+      currentList,
+      depositedAsset,
+      depositedAssetIndex,
+    ) => {
+      if (updatedDepositedAsset.symbol === depositedAsset.symbol) {
+        currentList[depositedAssetIndex] = updatedDepositedAsset;
+      }
+      return currentList;
+    }, depositedAssets);
+
+    dispatch(setDepositedAssetsAction(updatedDepositedAssets));
+  };
+};
+
+export const calculateLendingDepositTransactionEstimateAction = (
+  amount: number,
+  asset: AssetToDeposit,
+) => {
+  return async (dispatch: Dispatch, getState: GetState) => {
+    const { accounts: { data: accounts } } = getState();
+    const smartWalletAccount = findFirstSmartAccount(accounts);
+    if (!smartWalletAccount) return;
+
+    dispatch({ type: SET_CALCULATING_LENDING_DEPOSIT_TRANSACTION_ESTIMATE });
+
+    // may include approve transaction
+    const aaveDepositNeededTransactions = await getAaveDepositTransactions(
+      getAccountAddress(smartWalletAccount),
+      amount,
+      asset,
+    );
+
+    const estimateTransactions = aaveDepositNeededTransactions.map(({
+      to: recipient,
+      amount: value,
+      data,
+    }) => ({ recipient, value, data }));
+
+    const estimateTransaction: AccountTransaction = {
+      ...estimateTransactions[0],
+      sequentialTransactions: estimateTransactions.slice(1), // exclude first, take rest if exist
+    };
+
+    const estimate = await smartWalletService
+      .estimateAccountTransaction(estimateTransaction)
+      .catch(() => null);
+
+    dispatch({ type: SET_LENDING_DEPOSIT_TRANSACTION_ESTIMATE, payload: estimate });
+  };
+};
+
+export const calculateLendingWithdrawTransactionEstimateAction = (
+  amount: number,
+  depositedAsset: DepositedAsset,
+) => {
+  return async (dispatch: Dispatch, getState: GetState) => {
+    const { accounts: { data: accounts } } = getState();
+    const smartWalletAccount = findFirstSmartAccount(accounts);
+    if (!smartWalletAccount) return;
```

Comment:
No estimate will be provided, but Smart Wallet not being in wallet would cause more issues far before reaching this point. We keep adding this check because we assume there might be no Smart Wallet account (we had it in beta, then migration happened, etc.) and code validation fails, however, we can easily remove that assumption once key wallet is gone and we fully migrate to Smart Wallets.